src/CoreWCF.Primitives/src/CoreWCF/ServiceAuthorizationManager.cs

```diff
@@ -11,18 +15,52 @@ namespace CoreWCF
 {
     public class ServiceAuthorizationManager
     {
+        private bool _isCheckAccessCoreAsyncOverridden;
+        private bool _isCheckAccessAsyncWithSingleParameterOverridden;
+
+        public ServiceAuthorizationManager()
+        {
+            Type implementorType = GetType();
+            var methods = implementorType.GetMethods(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance);
+
+            var checkAccessCoreAsyncMethodInfo = methods.Single(static x => x.Name == nameof(CheckAccessCoreAsync));
+            var baseCheckAccessCoreAsyncMethodInfo = checkAccessCoreAsyncMethodInfo!.GetBaseDefinition();
+
+            _isCheckAccessCoreAsyncOverridden = baseCheckAccessCoreAsyncMethodInfo.DeclaringType != checkAccessCoreAsyncMethodInfo.DeclaringType;
+
+            var checkAccessAsyncWithSingleParameterMethodInfo = methods.SingleOrDefault(static x =>
+                x.Name == nameof(CheckAccessAsync) && x.GetParameters().Length == 1);
```

Comment:
If the implementation has another method with the same name and a single parameter, eg a helper method with the same name, how will this behave?

Reply:
I refined the logic of both flags